src/webgpu/api/validation/encoding/encoder_state.spec.ts

```diff
@@ -177,15 +177,20 @@ g.test('pass_end_none')
     }, endCount === 0);
   });
 
-g.test('pass_end_twice')
-  .desc('Test that ending a {compute,render} pass twice generates a validation error.')
+g.test('pass_end_twice,basic')
+  .desc(
+    'Test that ending a {compute,render} pass twice generates a validation error. The parent encoder (command encoder) can be either locked or open.'
+  )
   .paramsSubcasesOnly(u =>
     u //
       .combine('passType', ['compute', 'render'])
+      // Simply end twice, the parent encoder is open at that time. If the second pass end is in the middle of another pass, the parent encoder is locked. It should generate a validation error in either situation.
       .combine('endTwice', [false, true])
+      .combine('secondEndInAnotherPass', [false, true])
```

Comment:
nit: Just for completeness can we test both compute and render for this extra pass? (`false, 'compute', 'render'`)

Reply:
Good point. Done!